docs/testplan/multi_tier_fec_test.md

```diff
@@ -0,0 +1,23 @@
+# SONiC Switch FEC Test
+
+## Test Objective
+
+This test aims to verify that SONiC switch links properly respond to severe FEC errors by shutting down the operational status of affected ports.
+
+## Test Setup
+
+The test is designed to be topology-agnostic, meaning it does not assume or impose a specific network connection. The only requirement is that the DUT is fully connected to handle traffic loads. Plus all traffic generators in the testbed must be time synchronized.
+
+## Background
+
+Forward Error Correction (FEC) is an error control technique in data transmission where the transmitter sends redundant data, allowing the receiver to detect and correct a limited number of errors. If the number of erroneous packets exceeds the tolerance threshold, the optical module of the SONiC switch's ports will shut down the link. This test evaluates the FEC mechanism's ability to trigger a link-down event under severe error conditions.
+
+## Test Steps
+
+1. Retrieve the hardware SKU setting from config_DB to determine the DUT's port breakout mode. The breakout modes can be one of the following: 8x100Gbps, 4x200Gbps, 2x400Gbps, or 1x800Gbps, where the first number represents the number of logical ports per physical port, and the second number represents the speed of each logical port. Assume the DUT has X physical ports, each breaking out into Y logical ports.
```

Comment:
the breakout modes depend on the DUT capability. there are other devices running at 100G, 40G and etc.

Reply:
this step might not be needed, and we can simply get ask each logical port on the traffic generator to do this.